R/statistic-tree.R

```diff
@@ -170,22 +170,26 @@ dpTree <- setRefClass(
 )
 
 dpTree$methods(
+    # DO NOT USE
     initialize = function(mechanism, varType, variable, n, rng=NULL, gran, epsilon=NULL,
                           accuracy=NULL, imputeRng=NULL, percentiles=NULL, alpha=0.05, ...) {
         .self$name <- 'Differentially private binary tree'
-        .self$mechanism <- mechanism
-        .self$varType <- varType
+        .self$mechanism <- checkMechanism(mechanism, "mechanismLaplace")
+        .self$varType <- checkVariableType(varType, c('numeric', 'integer', 'logical', 'character'))
         .self$variable <- variable
-        .self$n <- checkNValidity(n)
-        .self$rng <- rng
-        .self$gran <- gran
-        .self$alpha <- alpha
+        .self$n <- checkN(n)
+        .self$rng <- checkRange(rng) # CHANGE
+        .self$gran <- checkN(gran, emptyOkay=TRUE) #should be positive whole number
+        .self$alpha <- checkNumeric(alpha)
         .self$sens <- 2 * log2(diff(rng) / gran + 1)
+        
+        checkVariableType(variable, "character")
+        
         if (is.null(epsilon)) {
-            .self$accuracy <- accuracy
+            .self$accuracy <- checkAccuracy(accuracy)
```

Comment:
why does this not have an "expectedLength = 1" parameter?

Reply:
because expectedLength defaults to 1.